core/src/main/scala/variant/Antichess.scala

```diff
@@ -103,3 +98,12 @@ case object Antichess
     // The pawn cannot attack a bishop or be attacked by a bishop
     val cannotAttackBishop = pawn.isLight != oppositeBishopLight
     InsufficientMatingMaterial.pawnBlockedByPawn(pawn, position) && cannotAttackBishop
+
+  private def allOnSameColourSquares(position: Position): Boolean =
+    (position.white.squares.forall(_.isLight) && position.black.squares.forall(_.isLight)) ||
+      (position.white.squares.forall(s => !s.isLight) && position.black.squares.forall(s => !s.isLight))
+
+  private def justOneKnightEach(position: Position): Boolean =
+    position.onlyKnights && position.byPiece(Color.White, Knight).count == 1 && position
+      .byPiece(Color.Black, Knight)
+      .count == 1
```

Comment:
```suggestion
    position.onlyKnights && position.white.count == 1 && position.black.count == 1
```
We know from the first condition that we only have knights so we only need to count white and black.